dynamodb-stream-eventbridge-fanout/src/main/java/no/unit/nva/doi/publisher/EventBridgeRetryClient.java

```diff
@@ -0,0 +1,67 @@
+package no.unit.nva.doi.publisher;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.eventbridge.EventBridgeClient;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequest;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsResponse;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsResultEntry;
+
+public class EventBridgeRetryClient {
+
+    private final EventBridgeClient eventBridge;
+    private final int maxAttempt;
+
+    private static final Logger logger = LoggerFactory.getLogger(EventBridgeRetryClient.class);
+
+    /**
+     * Constructor for EventBridgeRetryClient.
+     *
+     * @param eventBridge   eventBridge
+     * @param maxAttempt    maxAttempt
+     */
+    public EventBridgeRetryClient(EventBridgeClient eventBridge, int maxAttempt) {
+        this.eventBridge = eventBridge;
+        this.maxAttempt = maxAttempt;
+    }
+
+    /**
+     * Put events on EventBridge EventBus.
+     *
+     * @param request   request
+     * @return  list of PutEventsRequestEntry
+     */
+    public List<PutEventsRequestEntry> putEvents(final PutEventsRequest request) {
```

Comment:
this method is hard to read, needs refactoring.

Reply:
Agreed!